lib/audit/moab_to_catalog.rb

```diff
@@ -1,14 +1,19 @@
 ##
 # Method that will check the a single moab service disk for the existence of moabs in postgres database
 class MoabToCatalog
-  def self.check_moab_to_catalog_existence(storage_dir)
+  def self.check_moab_to_catalog_existence(storage_dir, expect_to_create=false)
     results = []
     MoabStorageDirectory.find_moab_paths(storage_dir) do |druid, path, _path_match_data|
       moab = Moab::StorageObject.new(druid, path)
       moab_current_version = moab.current_version_id
       moab_size = Stanford::StorageServices.object_size(druid)
-      po_handler = PreservedObjectHandler.new(druid, moab_current_version, moab_size)
-      results << po_handler.update_or_create
+      po_handler = PreservedObjectHandler.new(druid, moab_current_version, moab_size, storage_dir)
+      if PreservedObject.exists?(druid: druid)
+        results << po_handler.update
+      else
+        Rails.logger.error "druid: #{druid} expected to exist in catalog but was not found" unless expect_to_create
+        results << po_handler.create
```

Comment:
er ... don't we only create if "expect_to_create" is true?   o.w. we don't?

I guess I'm thinking when 
- we're seeding the db for the first time -- we're calling create
- we're auditing the moabs against the catalog -- we're calling update and we might be "expect_to_create" ... or we might not. ??

maybe we just need to leave the code like this and have a ticket/question for "will we always "expect_to_create" for moab_to_catalog?"   and a separate ticket (if we don't already have one) for "seed the catalog" that calls create.

Reply:
i think the idea is that we always create, but when we expect the object to be there, we log a warning.

so, in your first example case (initially seeding db), the flag would be true, and we don't log anything, because we expect to be creating everything from scratch.  in your second example (auditing against the catalog), we don't expect to create, because we expect ingest robots to have created the entry already, so we log a warning that we just found an unexpected inconsistency between the catalog and what's on disk.

i guess my current thinking is that the code for regular inventory audit and initial spin up is mostly the same, other than the expectation about whether the object is already in the catalog.  though, now that i think about it, if ingest messages about updates too, then maybe any inconsistency (object not in catalog OR object already in catalog but at the wrong version) results in a warning?

Reply:
F2F group discussion -- we agreed that there will be a "seed db" entry point into code that calls "create" and a separate "M2C existence check" entry point into code.  @jmartin-sul created appropriate tickets for that work, so this is now 👍 in the context of this PR.